libs/io/bbhnet/io/h5.py

```diff
@@ -39,7 +39,11 @@ def write_timeseries(
 
         for key, value in datasets.items():
             if len(value) != len(t):
-                raise ValueError("Length of t and data arrays don't match")
+                raise ValueError(
+                    "Length of data array:",
+                    key,
+                    "doesn't match the length of timeseries",
+                )

```

Comment:
this will be rendered better as a single string: `f"Length of data array '{key}' doesn't match the length of timeseries"`

Reply:
This change is incorporated